app/Services/OpenAIService.php

```diff
@@ -0,0 +1,122 @@
+            "Ukraine", "Belarus", "Lithuania", "Latvia", "Estonia", "Russia", "Ireland", "United Kingdom", "Turkey", "Cyprus",
+            "Malta", "Iceland", "Luxembourg", "Moldova", "Liechtenstein", "Andorra", "Monaco", "San Marino", "Vatican City",
+            "Armenia", "Azerbaijan", "Georgia", "Israel", "Japan", "New Zealand", "Philippines", "South Korea", "Singapore",
+            "Thailand", "Vietnam", "Bangladesh", "Bhutan", "Brunei", "Cambodia", "East Timor", "Fiji", "Kiribati", "Laos",
+            "Maldives", "Marshall Islands", "Micronesia", "Mongolia", "Myanmar", "Nauru", "Nepal", "Palau", "Papua New Guinea",
+            "Samoa", "Solomon Islands", "Sri Lanka", "Tonga", "Tuvalu", "Vanuatu", "Algeria", "Angola", "Benin", "Botswana",
+            "Burkina Faso", "Burundi", "Cabo Verde", "Cameroon", "Central African Republic", "Chad", "Comoros", "Congo",
+            "Djibouti", "Egypt", "Equatorial Guinea", "Eritrea", "Eswatini", "Ethiopia", "Gabon", "Gambia", "Ghana", "Guinea",
+            "Guinea-Bissau", "Ivory Coast", "Kenya", "Lesotho", "Liberia", "Libya", "Madagascar", "Malawi", "Mali", "Mauritania",
+            "Mauritius", "Morocco", "Mozambique", "Namibia", "Niger", "Nigeria", "Rwanda", "Sao Tome and Principe", "Senegal",
+            "Seychelles", "Sierra Leone", "Somalia", "South Africa", "South Sudan", "Sudan", "Tanzania", "Togo", "Tunisia", "Uganda",
+            "Zambia", "Zimbabwe", "Afghanistan", "Bahrain", "Iran", "Iraq", "Jordan", "Kuwait", "Lebanon", "Oman",
+            "Qatar", "Saudi Arabia", "Syria", "United Arab Emirates", "Yemen", "Albania", "Andorra", "Armenia", "Austria",
+            "Azerbaijan", "Belarus", "Belgium", "Bosnia and Herzegovina", "Bulgaria", "Croatia", "Cyprus", "Czech Republic",
+            "Denmark", "Estonia", "Finland", "France", "Georgia", "Germany", "Greece", "Hungary", "Iceland", "Ireland",
+            "Italy", "Kazakhstan", "Kosovo", "Latvia", "Liechtenstein", "Lithuania", "Luxembourg", "Malta", "Moldova",
+            "Monaco", "Montenegro", "Netherlands", "North Macedonia", "Norway", "Poland", "Portugal", "Romania", "Russia",
+            "San Marino", "Serbia", "Slovakia", "Slovenia", "Spain", "Sweden", "Switzerland", "Turkey", "Ukraine", "United Kingdom",
+            "Vatican City", "Argentina", "Belize", "Bolivia", "Brazil", "Chile", "Colombia", "Costa Rica", "Cuba", "Dominican Republic",
+            "Ecuador", "El Salvador", "Guatemala", "Guyana", "Haiti", "Honduras", "Jamaica", "Mexico", "Nicaragua", "Panama",
+            "Paraguay", "Peru", "Suriname", "Uruguay", "Venezuela", "Antigua and Barbuda", "Bahamas", "Barbados", "Dominica",
+            "Grenada", "Saint Kitts and Nevis", "Saint Lucia", "Saint Vincent and the Grenadines", "Trinidad and Tobago"];
+    }
+
+    public function importRulesForAllCities(bool $force): void
+    {
+        $cities = City::query()->whereHas("cityProviders")->orderBy("country_id")->get();
```

Comment:
Do we import rules for all cities or only with providers?

Reply:
Only with providers to save on OpenAI API